validator/src/key_utils.rs

```diff
@@ -0,0 +1,42 @@
+use std::collections::HashMap;
+
+use nimiq_bls::{CompressedPublicKey, KeyPair as BlsKeyPair};
+
+pub struct VotingKeys {
+    keys: HashMap<CompressedPublicKey, BlsKeyPair>,
+    current_key: BlsKeyPair,
+}
+
+impl VotingKeys {
+    pub fn new(keys: Vec<BlsKeyPair>) -> Self {
+        assert!(!keys.is_empty());
+        let mut key_hm = HashMap::new();
+        for key in &keys {
+            key_hm.insert(key.public_key.compress(), key.clone());
+        }
+        VotingKeys {
+            keys: key_hm,
+            current_key: keys.first().unwrap().clone(),
+        }
+    }
+
+    pub fn add_key(&mut self, key: BlsKeyPair) {
+        self.keys.insert(key.public_key.compress(), key);
+    }
+
+    pub fn get_current_key(&self) -> BlsKeyPair {
+        self.current_key.clone()
+    }
+
+    pub fn get_keys(&self) -> Vec<BlsKeyPair> {
+        self.keys.values().cloned().collect()
+    }
+
+    pub fn update_current_key(&mut self, public_key: &CompressedPublicKey) {
+        self.current_key = self
+            .keys
+            .get(public_key)
+            .expect("Invalid validator configuration: Expected voting key not available")
```

Comment:
This is the user-visible panic when the user starts the client without the correct key, but also when the key is rotated on-chain and an election block happens.

I'm not sure what the wanted behavior is here. It's probably good to quit the client if the correct signing key isn't available on startup, but I'm not sure what to do when the client is already running and has hit an election block.